src/odemis/acq/stream/_sync.py

```diff
@@ -1054,6 +1006,193 @@ def _runAcquisition(self, future):
             self._current_future = None
             self._acq_done.set()
 
+    def _startLeeches(self, rep_time, rep, tot_num):
+        """
+        A leech can be drift correction (dc) and/or probe/sample current (pca).
+        During the leech time the drift correction and/or the probe current measurements are conducted.
+        Start a counter np (= next pixel) until a leech is called.
+        Leech is called after a well specified number of acquired pixel positions.
+        :param rep_time (0<float): dwell time of one pixel
+        :param rep (int, int): 2D grid of pixel positions to be acquired
+        :param tot_num (int): total number of e-beam positions
+        :return:
+            leech_np (list of 0<int or None): for each leech, number of pixels before the leech should be
+                                              executed again. It's automatically updated inside the list.
+                                              (np = next pixels)
+            leech_time_ppx (float): extra time needed on average for a single pixel for all leches (s)
+        """
+
+        leech_np = []
+        leech_time = 0  # how much time leeches will cost
+        for l in self.leeches:
+            # TODO: store each np, as an approximation of how often it should
+            # be run, or just divide the estimated time by the number of
+            # total pixels, to compute the "bonus"
+            try:
+                leech_time += l.estimateAcquisitionTime(rep_time, (rep[1], rep[0]))
+                np = l.start(rep_time, (rep[1], rep[0]))  # np = next pixel = counter until execution
+            except Exception:
+                logging.exception("Leech %s failed to start, will be disabled for this acquisition", l)
+                np = None
+            leech_np.append(np)
+
+        # extra time needed on average for a single pixel for all leches (s)
+        leech_time_ppx = leech_time / tot_num  # s/px
+
+        return leech_np, leech_time_ppx
+
+    def _stopLeeches(self):
+        """
+        Stop the leeches after all pixels are acquired.
+        """
+        for l in self.leeches:
+            l.complete(self.raw)
+
+    def _waitForImage(self, start, rep_time):
+        """
+        Wait for the detector to acquire the image
+        :param start (0<float): start time for acquisition
+        :param rep_time (0<float): dwell time of one pixel
+        :return (bool): True if acquisition timed out
+        """
+
+        # A big timeout in the wait can cause up to 50 ms latency.
+        # => after waiting the expected time only do small waits
+
+        # wait for detector to acquire image
+        endt = start + rep_time * 3 + 5
+        timedout = not self._acq_complete[-1].wait(rep_time + 0.01)
+        if timedout:
+            logging.debug("Waiting more for rep")
```

Comment:
`rep` is actually really a bad name, let's make the log `Waiting more for detector %d`, and the other log `Got synchronized acquisition from detector %d`.